### PR TITLE
Add Contract Bytecode Size CI check

### PR DESCRIPTION
**Pull request by aureliusbtc**
### Summary by CodeRabbit

- New Feature: Implemented a new automated workflow to ensure the size of our project remains within acceptable limits. This will help maintain the performance and speed of our software.
- Documentation: Updated the README file for the `@synapsecns/contracts-core` package, enhancing the clarity of usage instructions and adding badges for easy access to documentation and version information.
- Test: Added a "size-check" job to our testing workflow, which will automatically check the size of the project during testing phases. This will help us identify and address any size-related issues promptly.

### CODE REVIEW

File: .github/workflows/forge-build-check.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Setup Node JS
+        uses: ./.github/actions/setup-nodejs
+
+      - name: Installing dependencies
```
Comment (trajan0x) [inline .github/workflows/forge-build-check.yml:24]: Not needed, setup-nodejs will take care of this

File: .github/workflows/forge-build-check.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        working-directory: ./packages/contracts-core
+        run: yarn install --immutable
```
Comment (trajan0x) [inline .github/workflows/forge-build-check.yml:26]: Also the --immutable option is deprecated

**Review by trajan0x — CHANGES_REQUESTED** [review]

File: .github/workflows/forge-build-check.yml (hunk 1 above)
Comment (aureliusbtc) [reply .github/workflows/forge-build-check.yml:24]: see CI committed by @trajan0x for which this action was based on, is this one wrong as well?

https://github.com/synapsecns/sanguine/blob/master/.github/workflows/foundry-tests.yml#L34

Comment (trajan0x) [reply .github/workflows/forge-build-check.yml:24]: yessir

File: .github/workflows/forge-build-check.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          forge build --sizes
+        working-directory: packages/contracts-core
+        continue-on-error: true # This will ensure the step doesn't fail, and we can check the exit code next
+
+      - name: Check exit code
+        run: |
+          if [ "$?" -eq "1" ]; then
+            echo "forge build --sizes failed!"
+            exit 1
+          fi
```
Comment (ChiTimesChi) [inline .github/workflows/forge-build-check.yml:44]: Can we omit this and let the job fail by having a failed `forge build --sizes`? The exit code check doesn't bring any extra granularity here imo.

**Comment by trajan0x** [comment]
consolidated in a96186127a76f8b4bcdc9f5f0e7733adf3b32eeb

**Pull request merged**